OpenRA.Game/Platform.cs

```diff
@@ -254,5 +255,20 @@ public static string ResolvePath(string path)
 
 			return path;
 		}
+
+		/// <summary>Allow the user to pick a path.</summary>
+		public static bool OpenDirectory(out string path)
+		{
+			var result = Dialog.FolderPicker();
```

Comment:
a. `ShowDirectoryOpenDialog`? `ShowFolderPicker`?

b. initialize path to null before opening the Dialog. not sure if it can raise exceptions.

c. what would happen on reentry? i.e. 2 mouse clicks causing two calls?

Reply:
its a native library. You cant click twice, not can it throw an exception. It returns a result which has either .IsError .IsCancelled or .IsOk set. So whatever goes wrong if cully captured before. All we get here it the info if it successfully selected something or not. And if its successful the selected path. As the app pauses completely till it returns from the dialog, i would vote against a Show... name, and more likely use a TryPickFolder. Just as a favour to name it by what it is used for, instead of what the user visualy sees. What the user sees is the underlying OS implementation. Theoretically this could open up a terminal, who knows. All we know is that it tells us if the user selected a folder, and if which it was.

Reply:
⚠️ ⚠️ ⚠️ !

Native deps are a problem: we target arm64 on Linux and macOS, and most nuget packages (including this one) don't support those.